lib/Core/Memory.h

```diff
@@ -382,6 +378,7 @@ class PersistentState : public ObjectState {
      * set.
      */
     UpdateList rootCauseLocations;
+    UpdateList pendingRootCauseLocations;
```

Comment:
Ah, I think I see what you've done here. You're waiting for the fence to clear the root cause, yes? So, in theory, you should be able to, for unfenced but flushed writes, to report both the write and the flush that were not fenced.

1. Can we do that easily?
2. Can you write this in a comment?

Reply:
1. No, we can't tell them which flush wasn't fenced because the pointer we write for a flush is `0`. We could possibly change that. Would storing something besides `0` as a result of a flush break the root cause tracking?
2. Yes, hadn't bothered cuz it was a last-minute fix.

Reply:
1. I don't think it would.